buildengine/engine.go

```diff
@@ -223,13 +221,41 @@ func (e *Engine) Dev(ctx context.Context, period time.Duration) error {
 }
 
 func (e *Engine) buildAndDeploy(ctx context.Context, replicas int32, waitForDeployOnline bool, modules ...string) error {
-	err := e.buildWithCallback(ctx, func(ctx context.Context, module Module) error {
-		return Deploy(ctx, module, replicas, waitForDeployOnline, e.client)
-	}, modules...)
-	if err != nil {
-		return err
+	if len(modules) == 0 {
+		modules = maps.Keys(e.modules)
 	}
-	return nil
+
+	deployQueue := make(chan Module, len(modules))
+	wg, ctx := errgroup.WithContext(ctx)
+	wg.SetLimit(runtime.NumCPU())
+
+	wg.Go(func() error {
+		defer close(deployQueue)
+
+		return e.buildWithCallback(ctx, func(ctx context.Context, module Module) error {
+			select {
+			case deployQueue <- module:
+				return nil
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+		}, modules...)
+	})
+
+	for i := 0; i < len(modules); i++ {
+		wg.Go(func() error {
+			for module := range deployQueue {
```

Comment:
This should be a for loop with a select inside, that checks `<-ctx.Done()` as the other Go call does.